src/Jobs/AlgoliaReindexAllJob.php

```diff
@@ -104,49 +104,58 @@ public function process()
             return;
         }
 
+        $algoliaService = Injector::inst()->create(AlgoliaService::class);
         $task = new AlgoliaReindex();
 
         $batchSize = $task->config()->get('batch_size');
         $batching = $this->config()->get('use_batching');
 
         foreach ($remainingChildren as $class => $ids) {
-            $take = array_slice($ids, 0, $batchSize);
-            $this->indexData[$class] = array_slice($ids, $batchSize);
+            foreach ($algoliaService->indexes as $indexName => $index) {
+                $classes = (isset($index['includeClasses'])) ? $index['includeClasses'] : [];
 
-            if (!empty($take)) {
-                $this->currentStep += count($take);
-                $errors = [];
+                if (!in_array($class, $classes)) {
+                    continue;
+                }
 
```

Comment:
Not sure why this diff looks so ugly, all that changed was I wrapped the meat of this in
```php
foreach ($algoliaService->indexes as $indexName => $index) {
  $classes = (isset($index['includeClasses'])) ? $index['includeClasses'] : [];

  if (!in_array($class, $classes)) {
    continue;
  }
```
And changed the indexItems to take $indexName instead of $class